src/pages/ReimbursementAccount/BusinessInfo/substeps/PhoneNumberBusiness.tsx

```diff
@@ -43,7 +43,7 @@ function PhoneNumberBusiness({reimbursementAccount, onNext, isEditing}: PhoneNum
 
     const handleSubmit = useReimbursementAccountStepFormSubmit({
         fieldIds: STEP_FIELDS,
-        isEditing,
+        isEditing: true,
```

Comment:
Just to clarify, I haven't asked for new videos. Thanks for uploading them, as I assume that they contain some changes that you considered important.

Reply:
> Don't we leave the isEditing prop of PhoneNumberBusiness unused?

Ok, nevermind. It is used here:

https://github.com/Expensify/App/blob/59830b8f8138d918be3f4d6b1a121d694762857b/src/pages/ReimbursementAccount/BusinessInfo/substeps/PhoneNumberBusiness.tsx#L53

Reply:
> This works like when we set the prop to true, we reduce the values and hence setting it to true does our work over here

I'm sorry, I can't see the answer in these words, so I'll rephrase my question.

The solution we agreed on is to trim the phone number from excessive whitespace.

On the technical level, we set "is editing" to "true", which triggers `reduce` operation which process some JS objects. Where's the trimming?

I'm not saying this is not working, but when your solution is to "water the flowers" and you solve it by doing "feed the cat: true", you should explain how one achieves the other.

Reply:
There are two ways forward here. We either...

- add a good comment explaining how (and why) we water the flowers by feeding a cat
- or find another solution

Reply:
Okey, let's do this:

```ts
/**
 * Hook for handling submit method in ReimbursementAccount substeps.
 * Typically, we want to save the draft values after the user confirmed them in the editing mode.
 * @param formId - ID for particular form
 * @param shouldSaveDraft - should the form draft values be saved?
 * @param onNext - callback
 * @param fieldIds - field IDs for particular step
 */
export default function useReimbursementAccountStepFormSubmit({
    formId = ONYXKEYS.FORMS.REIMBURSEMENT_ACCOUNT_FORM,
    shouldSaveDraft, // renamed from `isEditing`. Update all usages.
    onNext,
    fieldIds,
}: UseReimbursementAccountStepFormSubmitParams) {
    // No changes
}
```

Note the suggested comment change.

Use in `PhoneNumberBusiness.tsx`:

```ts
    const handleSubmit = useReimbursementAccountStepFormSubmit({
        fieldIds: STEP_FIELDS,
        // Save the form draft after this step. The submitted value is "prepared" (trimmed), which ensures that the
        // phone number won't be displayed with excessive whitespace later.
        shouldSaveDraft: true,
        onNext,
    });
```

This will reflect our intention without abusing the `isEditing` property.

Reply:
I suggest calling it like...

```ts
shouldSaveDraft: isEditing,
```

in most places.

This way, we will keep the meaning of the `isEditing` property—the original solution kind of "hijacked" it and smelled hacky in my opinion.